### PR TITLE
[Warnings/Errors] Design and accessibility improvements

### LINKED ISSUE #568: [Design Review] Warnings
- [x] Content review
- [x] Fix: Table border on last page of paginated tables
- [x] Fix: Make "Upload a new file" a secondary button (now that Download report is a primary button)
- [x] Task: Run an accessibility review
  - [x] Table - labeled element should not be nested in the label (or at least the `label[for=...]` should match between the two (DSR?)
  - [x] StepIndicator - Remove 'role="grid"'
  - [x] FilingFieldLinks - Button in list needs to be wrapped in ListItem

Post pre-clearance
- Always enable `Save and continue` on page
  - WIP PR: https://github.com/cfpb/sbl-frontend/pull/552

### PR DESCRIPTION
**Pull request by meissadia**
Close #568 

## Changes
- [Accessibility] Fix labelling issue in DSR Table/Pagination 
  - Dependent upon https://github.com/cfpb/design-system-react/pull/356
- `Upload a new file` now a secondary button ([should be a link but I couldn't get the styling overrides to play nice with one another](https://github.com/cfpb/sbl-frontend/issues/575))
- Update primary button label: `Download Report (CSV)`
- Remove green header border on Filing pages

## Screenshots
|Change|Screenshot|
|---|---|
|Button label (Warnings)|Screenshot 2024-05-31 at 4 04 11 PM|
|Button label (Errors)|Screenshot 2024-05-31 at 4 09 02 PM|
|Remove green header border on filing|Screenshot 2024-06-03 at 9 54 54 AM|
|Accessibility for pagination: input's `for` label aligns with table id|Screenshot 2024-06-03 at 9 53 03 AM|

### CODE REVIEW

**Review by billhimmelsbach — COMMENTED** [review]
Hey @meissadia, when you run this with ANDI, do you still see errors with the label?

**Comment by billhimmelsbach** [comment]
If you end up needing it @meissadia, here's a PR with the latest DSR that can be merged into your branch:
https://github.com/cfpb/sbl-frontend/pull/632

**Comment by meissadia** [comment]
> If you end up needing it @meissadia, here's a PR with the latest DSR that can be merged into your branch: #632

@billhimmelsbach Many thanks for this!  Don't know why I couldn't get the dependency to update...

File: src/pages/Filing/FilingApp/FilingFieldLinks.tsx [hunk 1]
```diff
@@ -20,21 +21,37 @@ function FilingFieldLinks({
...
-          label='Download report (CSV)'
-          // eslint-disable-next-line @typescript-eslint/no-misused-promises
-          onClick={onHandleDownloadClick}
-        />
-        <ListLink href={`/filing/${filingPeriod}/${lei}/upload`}>
-          Upload a new file
-        </ListLink>
+        <ListItem>
+          <Button
+            label='Download report (CSV)'
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Filing/FilingApp/FilingFieldLinks.tsx]: Code updated, screenshots not updated.

**Review by billhimmelsbach — APPROVED** [review]
Looks great! Thanks for continuing to update this one!

**Pull request merged**